Client/Frontend/UserContent/UserScripts/WalletSolanaProvider.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function($Object) {
+  if (window.isSecureContext) {
+    function post(method, payload, completion) {
+      return new Promise((resolve, reject) => {
+        webkit.messageHandlers.$<handler>.postMessage({
```

Comment:
I might be abit confused here, but the usage of postMessage makes me think that there may be some unexpected behavior here if the frame directly calls this messageHandler via a postMessage rather than through our communication layer. Is that a possibility which the way this is written?

Reply:
I can answer this, I think, since we just completed the JS security review.  It's technically possible to call postMessage directly, however, to do this, a webpage has to know the message handler's name and the security token, which is randomized every time the browser is started. And after the latest JS rework by @Brandon-T it's not possible (as far as we know) to leak these values.

Reply:
Ok good to know. That makes sense and isn't that much different from most other programming language designs then. Sounds like basically an object capability model with the slight difference being that this assumption can be broken with the ability to directly inspect memory (rather than write/use memory like most overflow style attacks) which is still quite hard.

Thanks for the context on how these work better as well!